fuzzing/private/oss_fuzz/package.bzl

```diff
@@ -0,0 +1,71 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Rule for packaging fuzz tests in the expected OSS-Fuzz format."""
+
+load("//fuzzing/private:binary.bzl", "CcFuzzingBinaryInfo")
+
+def _oss_fuzz_package_impl(ctx):
+    output_archive = ctx.actions.declare_file(ctx.label.name + ".tar")
+    binary_info = ctx.attr.binary[CcFuzzingBinaryInfo]
+
+    action_inputs = [binary_info.binary_file]
+    if binary_info.corpus_dir:
+        action_inputs.append(binary_info.corpus_dir)
+    if binary_info.dictionary_file:
+        action_inputs.append(binary_info.dictionary_file)
+    ctx.actions.run_shell(
+        outputs = [output_archive],
+        inputs = action_inputs,
+        command = """
+            declare -r STAGING_DIR="$(pwd)/{output}.staging"
+            mkdir "$STAGING_DIR"
+            ln -s "$(pwd)/{binary_path}" "$STAGING_DIR/{base_name}"
+            if [[ -n "{corpus_dir}" ]]; then
+                pushd "{corpus_dir}" >/dev/null
+                zip --quiet -r "$STAGING_DIR/{base_name}_seed_corpus.zip" ./*
+                popd >/dev/null
+            fi
+            if [[ -n "{dictionary_path}" ]]; then
+                ln -s "$(pwd)/{dictionary_path}" "$STAGING_DIR/{base_name}.dict"
+            fi
+            tar -chf "{output}" -C "$STAGING_DIR" .
+        """.format(
```

Comment:
@inferno-chromium @oliverchang Let me know if you notice anything wrong with this packaging structure.

Reply:
The seed_corpus and dict paths look good, but would it make sense support generating options? 

e.g. letting people support specifying options in the rule that would generate something like: https://github.com/google/oss-fuzz/blob/master/projects/harfbuzz/hb-draw-fuzzer.options

Reply:
There are currently bazel projects which rely on .options files being copied to $OUT. e.g. https://github.com/google/oss-fuzz/blob/c1f1afe5300dc039b8e0988b4d8c1bee77fc7599/projects/grpc/build.sh#L127

I suppose they can still do this manual step/workaround to copy manually, but it would be nice if this was supported out of the box if we want to migrate these projects.

Reply:
Oh, I wasn't aware of these, thanks! I briefly scanned their options files, and they seem to set libfuzzer options such as the maximum input size. It looks like these would be useful for all the consumers of the fuzz target rule, including the launcher rule.

I filed #103 and #104 for this so I can address it more systematically and keep this PR focused on the basic OSS-Fuzz integration. This seems to be indeed a blocker for the broader OSS-Fuzz integration.